modules/tinymce/src/themes/silver/main/ts/ui/header/InlineHeader.ts

```diff
@@ -140,6 +176,22 @@ export const InlineHeader = (
     });
   };
 
+  const restoreAndGetCompleteOuterContainerWidth = (): Optional<number> => {
+    /*
+    Editors can be placed so far to the right that their left position is beyond the window width. This causes problems with flex-wrap. To solve this, set a width style on the container.
+    Natural width of the container needs to be calculated first.
+    */
+    if (!useFixedToolbarContainer) {
+      Css.set(mainUi.outerContainer.element, 'position', 'absolute');
+      Css.set(mainUi.outerContainer.element, 'left', '0px');
+      Css.remove(mainUi.outerContainer.element, 'width');
+      const w = Width.getOuter(mainUi.outerContainer.element);
```

Comment:
This is a fairly expensive process, particularly if we do it a lot while horizontal scrolling, and the results won't change very often. Is there a way to cache the value?

Reply:
We probably could cache it, but the issue is knowing for sure it's updated frequently enough, and that we don't end up with the wrong value.

Reply:
Yeah we'd need some way to compare what we have to the new state of the UI and determine whether it needs updating.

A simple fix is don't set a width when one isn't needed - that will reduce the impact of scrolling that doesn't trigger the bug (since you then won't be removing the width and repositioning the toolbar constantly during scroll).

From there, we could do things like reduce how much we measure unless the right side of the toolbar is within some area of pixels on the right side of the viewport.